plaso/parsers/sqlite_plugins/windows_timeline.py

```diff
@@ -0,0 +1,218 @@
+    appid_entries_string = str(self._GetRowValue(query_hash, row, 'AppId'))
+
+    # Both Payload and AppId are serialized as JSON blobs in the DB
+    payload = json.loads(payload_json_string)
+    appid_entries = json.loads(appid_entries_string)
+
+    # Attempt to populate the package_id field by checking each of these fields
+    # in the AppId json
+    package_id_locations = ['packageId', 'x_exe_path', 'windows_win32',
+                            'windows_universal', 'alternateId']
+    for location in package_id_locations:
+      for entry in appid_entries:
+        if entry['platform'] == location and entry['application'] != '':
+          event_data.package_id = entry['application']
+          break
+        # We haven't found this field, or the entry wasn't populated. Fall
+        # back to the next option
+      if event_data.package_id != None:
+        # package_id has been populated and we're done
+        break
+
+    if 'description' in payload:
+      event_data.description = payload['description']
+    else:
+      event_data.description = ''
+
+    if 'appDisplayName' in payload and payload['appDisplayName'] != '':
+      event_data.application_display_name = payload['appDisplayName']
+    elif 'displayText' in payload and payload['displayText'] != '':
+      # Fall back to displayText if appDisplayName isn't available
+      event_data.application_display_name = payload['displayText']
+
+    timestamp = self._GetRowValue(query_hash, row, 'StartTime')
+    date_time = dfdatetime_posix_time.PosixTime(timestamp=timestamp)
+    event = time_events.DateTimeValuesEvent(
+        date_time, definitions.TIME_DESCRIPTION_CREATION)
+    parser_mediator.ProduceEventWithEventData(event, event_data)
+
+  def ParseUserEngagedRow(
+      self, parser_mediator, query, row, **unused_kwargs):
+    """Parses a timeline row that describes a user interacting with an app.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+        and other components, such as storage and dfvfs.
+      query (str): query that created the row.
+      row (sqlite3.Row): row.
+    """
+    query_hash = hash(query)
+
+    event_data = WindowsTimelineUserEngagedEventData()
+    event_data.package_id = self._GetRowValue(query_hash, row, 'PackageName')
+
+    # The payload is serialized as a JSON blob in the DB
+    payload_json_string = str(self._GetRowValue(query_hash, row, 'Payload'))
+    payload = json.loads(payload_json_string)
+
+    if 'reportingApp' in payload:
+      event_data.reporting_app = payload['reportingApp']
+    if 'activeDurationSeconds' in payload:
+      event_data.active_duration_seconds = int(payload['activeDurationSeconds'])
+
+    timestamp = self._GetRowValue(query_hash, row, "StartTime")
+    date_time = dfdatetime_posix_time.PosixTime(timestamp=timestamp)
+    event = time_events.DateTimeValuesEvent(
+        date_time, definitions.TIME_DESCRIPTION_CREATION)
```

Comment:
Is creation time the right timestamp definition here? Isn't this execution time, or something similar?